app/models/mane_preferred_transcript.rb

```diff
@@ -0,0 +1,51 @@
+class ManePreferredTranscript
+  attr_reader :variant
+
+  def initialize(variant_id)
+    @variant = Variant.find(variant_id)
+  end
+
+  def mane_preferred_transcript
+    Rails.cache.fetch(cache_key(variant), expires_in: 24.hours) do
+      if mane_preferred_ref_seq = get_mane_preferred_ref_seq(variant)
+        transcripts = get_allele_registry_transcripts(variant)
+        transcripts.find{ |t| t['hgvs'][0].start_with?(mane_preferred_ref_seq) }
+      end
+    end
+  end
+
+  def get_allele_registry_transcripts(variant)
+    allele_response = make_allele_request(variant.allele_registry_id)
+    JSON.parse(allele_response)['transcriptAlleles']
+  end
+
+  def get_mane_preferred_ref_seq(variant)
+    gene_response = make_gene_request(variant.gene.name)
+    JSON.parse(gene_response)[0]['externalRecords']['MANEPrefRefSeq']['id'] rescue nil
+  end
+
+  private
+  def make_gene_request(gene_name)
+    Scrapers::Util.make_get_request(gene_url(gene_name))
+  rescue StandardError
+    {}
```

Comment:
I believe this will break when the return value is passed to `JSON.parse` which expects a string. I would either return a string `{}` or do the parsing in this method and catch either error here.

Reply:
I just pushed a commit to fix this.